pinot-query-planner/src/main/java/org/apache/calcite/rel/rules/PinotJoinExchangeNodeInsertRule.java

```diff
@@ -65,7 +65,7 @@ public void onMatch(RelOptRuleCall call) {
 
     if (joinInfo.leftKeys.isEmpty()) {
       // when there's no JOIN key, use broadcast.
-      leftExchange = LogicalExchange.create(leftInput, RelDistributions.SINGLETON);
+      leftExchange = LogicalExchange.create(leftInput, RelDistributions.RANDOM_DISTRIBUTED);
       rightExchange = LogicalExchange.create(rightInput, RelDistributions.BROADCAST_DISTRIBUTED);
```

Comment:
Not sure I follow this change. If we are doing a BROADCAST_EXCHANGE on one side, why do we need to insert any EXCHANGE on the other side ? May be I am missing something.

Reply:
it is the least restrictive exchange needed --> e.g. if right side has already broadcast the data; left side can be arbitrarily exchange its data to join stage.

- previously we can allow singleton here b/c we want to optimize local-to-local transfer without ser/de or grpc hopping;
- however with the new singleton stage, we might not be able to apply this optimization thus we relax the exchange to random

Reply:
Discussed offline with @walterddr. This is sort of faking an exchange.

My question was from the perspective that left side does not need data movement 

```
Leaf stage - scan -> filter -> project right side
BROADCAST_EXCHANGE right side
Intermediary stage to run JOIN (can be run on same set of servers where no data movement needed for left side)
```

Since the intermediary stage needs to run the new operators and there is no mechanism to wait currently, we need to do it this way.